tokio/src/macros/select.rs

```diff
@@ -562,41 +595,65 @@ macro_rules! select {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if true => $h, } $($r)*)
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if true => $h, } $($r)*)
     };
     (@ { start=$start:expr; ( $($s:tt)* ) $($t:tt)* } $p:pat = $f:expr, if $c:expr => $h:block $($r:tt)* ) => {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if $c => $h, } $($r)*)
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if $c => $h, } $($r)*)
     };
     (@ { start=$start:expr; ( $($s:tt)* ) $($t:tt)* } $p:pat = $f:expr => $h:block $($r:tt)* ) => {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if true => $h, } $($r)*)
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if true => $h, } $($r)*)
     };
     (@ { start=$start:expr; ( $($s:tt)* ) $($t:tt)* } $p:pat = $f:expr, if $c:expr => $h:expr ) => {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if $c => $h, })
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if $c => $h, })
     };
     (@ { start=$start:expr; ( $($s:tt)* ) $($t:tt)* } $p:pat = $f:expr => $h:expr ) => {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if true => $h, })
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if true => $h, })
     };
     (@ { start=$start:expr; ( $($s:tt)* ) $($t:tt)* } $p:pat = $f:expr, if $c:expr => $h:expr, $($r:tt)* ) => {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if $c => $h, } $($r)*)
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if $c => $h, } $($r)*)
     };
     (@ { start=$start:expr; ( $($s:tt)* ) $($t:tt)* } $p:pat = $f:expr => $h:expr, $($r:tt)* ) => {
-        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) $p = $f, if true => $h, } $($r)*)
+        $crate::select!(@{ start=$start; ($($s)* _) $($t)* ($($s)*) #[cfg(all())] $p = $f, if true => $h, } $($r)*)
     };
 
     // ===== Entry point =====
 
-    (biased; $p:pat = $($t:tt)* ) => {
-        $crate::select!(@{ start=0; () } $p = $($t)*)
+    (biased; $( #[cfg($meta:meta)] )? $p:pat = $($t:tt)* ) => {
+        $crate::select!(@{ start=0; () } $( #[cfg($meta)] )? $p = $($t)*)
     };
 
-    ( $p:pat = $($t:tt)* ) => {
+    ( $( #[cfg($meta:meta)] )? $p:pat = $($t:tt)* ) => {
         // Randomly generate a starting point. This makes `select!` a bit more
         // fair and avoids always polling the first future.
-        $crate::select!(@{ start={ $crate::macros::support::thread_rng_n(BRANCHES) }; () } $p = $($t)*)
+        $crate::select!(@{ start={ $crate::macros::support::thread_rng_n(BRANCHES) }; () } $( #[cfg($meta)] )? $p = $($t)*)
```

Comment:
As the code is now, the random number will no longer pick the branches with equal probability.

Reply:
Why?

Reply:
To illustrate my point, consider if we have three branches:
```Rust
select! {
    #[cfg(false)]
    _ = branch_0() => {},
    _ = branch_1() => {},
    _ = branch_2() => {},
}
```
Then we have `BRANCHES` equal to 3 and the random number generator will pick either 0, 1 or 2 at equal probability. However, if the generator chooses a 0, then that's turned in to a 1 by the continue. This makes option 1 twice as likely as option 2.

Reply:
Ah true. I didn't thought about that. What should I do for this?

Reply:
I don't really know what to do about it. I don't think there is an easy fix.